attachment_image_resize/models/ir_attachment.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2024 Quartile
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+IMAGE_TYPES = ["image/png", "image/jpeg", "image/bmp", "image/tiff"]
+
+
+class IrAttachment(models.Model):
+    _inherit = "ir.attachment"
+
+    resize_done = fields.Boolean()
+
+    def _postprocess_contents(self, values):
+        self = self.with_context(resize_target_model=values.get("res_model"))
+        return super()._postprocess_contents(values)
+
+    @api.model
+    def _cron_resize_attachment_image(self, limit):
+        model_list = [
+            rec.model
+            for rec in self.env["ir.model"].search(
+                [("attachment_image_max_resolution", "!=", False)]
+            )
+        ]
```

Comment:
To be a a bit more concise.
```suggestion
        model_list = self.env["ir.model"].search(
                [("attachment_image_max_resolution", "!=", False)]
            ).mapped("model")
```